app/repositories/claim-repository.js

```diff
@@ -68,23 +67,20 @@ export const updateClaimByReference = async (data) => {
       returning: true
     })
 
-    const updatedRows = result[0] // Number of affected rows
-    const updatedRecords = result[1] // Assuming this is the array of updated records
-    const sbi = data.sbi
+    const updatedRecord = result[1][0]
 
-    for (let i = 0; i < updatedRows; i++) {
-      const updatedRecord = updatedRecords[i]
+    if (updatedRecord) {
       await raiseClaimEvents({
         message: 'Claim has been updated',
         claim: updatedRecord.dataValues,
+        note,
         raisedBy: updatedRecord.dataValues.updatedBy,
         raisedOn: updatedRecord.dataValues.updatedAt
-      }, sbi)
+      }, data.sbi)
     }
```

Comment:
else should throw some kind of exception here? Something must've gone wrong if no updatedRecord was present

Reply:
Think I fell into the trap of making the tests pass. Removed the check and accompanying test, as you say, if an attempt is made to update a record that doesn't exist it will error, and we will log that in the existing catch.